worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -193,6 +193,47 @@ function applyActionDetails(document, actionDetails, terminateOnFailure) {
     }
 }
 
+function onAfterProcessDocument(e){
+    haveFailuresChanged(e.document);
+}
+
+function haveFailuresChanged(document) {
+    if (document.getFailures().isChanged()) {
+
+        var listOfFailures = new java.util.ArrayList();
+        document.getFailures().stream().forEach(function (failure) {
+            listOfFailures.add(failure);
+        });
+
+        document.getFailures().reset();
+
+        var listOfOriginalFailures = new java.util.ArrayList();
+        document.getFailures().stream().forEach(function (failure) {
+            listOfOriginalFailures.add(failure);
+        });
+
+        var newListOfFailures = new java.util.ArrayList();
+        for each (var f in listOfFailures) {
+            if (!listOfOriginalFailures.contains(f)) {
+                var message = {
+                    id: f.getFailureId(),
+                    stack: f.getFailureStack(),
+                    description: {
+                        source: document.rootDocument.getField("CAF_WORKFLOW_ACTION").getStringValues().get(0),
+                        version: document.getTask().getService(com.hpe.caf.api.worker.WorkerTaskData.class).getSourceInfo().getVersion(),
+                        workflowName: document.getField("CAF_WORKFLOW_NAME").getStringValues().get(0),
+                        originalDescription: f.getFailureMessage(),
+                        stack: f.getFailureStack()
+                    }
+                };
+                document.getFailures().add(f.getFailureId(), JSON.stringify(message));
+                newListOfFailures.add(JSON.stringify(message));
+            }
+        }
+        document.getField("FAILURES").add(newListOfFailures);
```

Comment:
You're adding the array as one field value.  Drop `newListOfFailures` and replace line 30 with:

    document.getField("FAILURES").add(JSON.stringify(message));

That way each failure will be a separate field value.